tests/fiaas_deploy_daemon/crd/test_crd_watcher.py

```diff
@@ -100,7 +128,157 @@ def status_get(self):
             m.side_effect = NotFound
             yield m
 
-    def test_creates_custom_resource_definition_if_not_exists_when_watching_it(self, get, post, crd_watcher, watcher):
+    def test_creates_apiextensions_v1_crd_if_not_exists_when_watching_it(self, get, post, apiextensions_v1_crd_watcher_with_syncer,
```

Comment:
Since CRD create/update now happens in CrdResourcesSyncerApiextensionsV1 and CrdResourcesSyncerApiextensionsV1Beta1, what do you think about changing this test and `test_creates_apiextensions_v1beta1_crd_if_not_exists_when_watching_it` to test only the `CrdResourcesSyncer$version.update_crd_resources` and move the tests into separate files for each class/api version? I think it could make the tests a little simpler and would remove the coupling between CrdWatcher and CrdResourcesSyncer in the tests - CrdWatcher tests could use a mock CrdResourcesSyncer and just assert `update_crd_resources` is called when `_watch` is called

Reply:
It would also be nice to add a test case for when the CRDs already exist, as this PR changes the behavior to always updating the CRDs